### PR TITLE
Release ExoJAX 2.6.0

### PR DESCRIPTION
**Pull request by HajimeKawahara**
## Summary

Prepare ExoJAX 2.6.0 for release from `v2.6` into `master`. See [here](https://github.com/HajimeKawahara/exojax/milestone/24?closed=1) for the v2.6.0 update since v2.5.0.

- source function methods
- direct Limb darkening
- precomputed CKD
- vr_array warning
- DiffGrid (Hermite interpolation of opacity grid with value and its inverse temperature derivative) 

 ## Bug fixes

  - #736 (https://github.com/HajimeKawahara/exojax/pull/736) Corrected the hydrostatic gravity profile to follow
    inverse-square scaling and improved dtype-rounding tolerance in Diffgrid boundary handling.

  - #737 (https://github.com/HajimeKawahara/exojax/pull/737), #761 (https://github.com/HajimeKawahara/exojax/pull/761)
    updated rotkernel to use the standard quadratic limb-darkening law, retained the previous convention as
    rotkernel_Gimenez, corrected JAX derivatives at the kernel boundary, and made `vsini=0` an AD-safe identity.

  - #738 (https://github.com/HajimeKawahara/exojax/pull/738) Fixed self-generated CKD tables to evaluate the full
    Cartesian temperature–pressure grid, restoring their temperature dependence. Previously generated tables should be
    regenerated.

  - #739 (https://github.com/HajimeKawahara/exojax/pull/739) Fixed pressure-unit conversion and condensate-density
    handling in the Ackerman–Marley cloud model, correcting drag regimes and inferred particle radii.

  - #740 (https://github.com/HajimeKawahara/exojax/pull/740) Fixed partition-function ratio axes in atomic
    OpaDirect.xsmatrix, enabling VALD and Kurucz calculations when the numbers of lines and atmospheric layers differ.

  - #741 (https://github.com/HajimeKawahara/exojax/pull/741) Fixed source-function scaling and bottom-to-top layer
    ordering in Opart flux adding, bringing emission and reflection results into agreement with the canonical batch
    solver.

  - #742 (https://github.com/HajimeKawahara/exojax/pull/742), #759 (https://github.com/HajimeKawahara/exojax/pull/759)
    fixed ArtAbsPure surface optical-depth integration in both line-by-line and CKD solvers. Partial layers now use
    linear-pressure fractions, including exact pressure boundaries and top-layer surfaces.

  - #743 (https://github.com/HajimeKawahara/exojax/pull/743) Fixed Direct and MODIT opacity calculations for all-isotope
    HITRAN/HITEMP databases (isotope=0) by applying line-specific partition-function ratios.

  - #744 (https://github.com/HajimeKawahara/exojax/pull/744) Stabilized linsap emission coefficients at zero and small
    optical depth, restoring accurate float32 behavior and finite automatic derivatives.

  - #745 (https://github.com/HajimeKawahara/exojax/pull/745), #747 (https://github.com/HajimeKawahara/exojax/pull/747),
    #757 (https://github.com/HajimeKawahara/exojax/pull/757), and #762 (https://github.com/HajimeKawahara/exojax/pull/762)
    stabilized pure-absorption and Toon two-stream transfer across zero, optically thin, extreme, and thick
    conservative-scattering limits, preserving physical float32 thermal emission, finite fluxes, and JAX derivatives.

  - #746 (https://github.com/HajimeKawahara/exojax/pull/746) Corrected the variable-sampling Gaussian response velocity
    coordinate to use the logarithmic wavenumber ratio, preventing NaN kernels over wide spectral ranges.

  - #748 (https://github.com/HajimeKawahara/exojax/pull/748) Fixed float32 lognormal Mie-grid sampling for narrow
    particle-size distributions, preventing cancellation-induced NaNs and zeroed Mie coefficients.

  - #749 (https://github.com/HajimeKawahara/exojax/pull/749) Fixed PreMODIT single-broadening initialization with
    one-element NumPy grids, which previously raised an IndexError.

  - #750 (https://github.com/HajimeKawahara/exojax/pull/750), #755 (https://github.com/HajimeKawahara/exojax/pull/755)
    fixed cloud-radius calculations and opacity propagation with layer-dependent Kzz, correctly broadcasting per-layer
    cloud factors along the spectral axis while retaining scalar support.

  - #751 (https://github.com/HajimeKawahara/exojax/pull/751), #758 (https://github.com/HajimeKawahara/exojax/pull/758)
    restored atomic MODIT VALD opacity paths by correcting matrix dispatch and species-axis mapping, repairing stale
    atomic I/O references, and preserving the reference temperature.

  - #752 (https://github.com/HajimeKawahara/exojax/pull/752), #756 (https://github.com/HajimeKawahara/exojax/pull/756)
    fixed uncovered CKD bands by using the dtype's smallest positive normal opacity floor, keeping empty bands
    effectively transparent in float32 radiative transfer. Tables generated with an older floor should be regenerated.

  - #753 (https://github.com/HajimeKawahara/exojax/pull/753) Stabilized float32 Planck-function values and gradients in
    cold, high-wavenumber regimes, preserving positive radiances and finite nonzero derivatives.

  - #754 (https://github.com/HajimeKawahara/exojax/pull/754), #760 (https://github.com/HajimeKawahara/exojax/pull/760)
    corrected direct Mie wavelength units and restored scalar and matrix PyMieScatt execution with current SciPy;
    precomputed Mie grids are unaffected.

### others

- refactorings/doc
- add the v2.6 release notes and align Python 3.10–3.13 metadata, runtime dependencies, packaging, CI, and Read the Docs configuration
- restore the deprecated `exojax.spec` import compatibility layer and repair stale compatibility shims
- restore the VALD PyTables cache backend as the default, retain the `pandas` alias and optional `vaex` backend, and fix `gpu_transfer=False`
- correct the WASP-39b inference comment; #642 remains open outside the v2.6 milestone
- temporarily allow inherited Sphinx warnings; follow-up is tracked in #729

### Updates merged from `develop` (2026-08-13)

Merged the latest `develop` through `b3f84197` into `v2.6` in `f3e935b0`.

- add pressure-aligned DiffGrid opacity interpolation and expose `OpaDiffgrid`
- add DiffGrid archive I/O, validation contracts, and precomputation utilities
- add the DiffGrid NUTS retrieval tutorial and user guides
- add a GPU benchmark comparing DiffGrid and PreMODIT in NUTS-like workloads
- add DiffGrid API, core, I/O, and grid-mode regression tests
- constrain NumPy to `<2.5` until RADIS supports NumPy 2.5
- add Pillow and IPython to the documentation build requirements

### Validation

- `JAX_PLATFORMS=cpu PYTHONPATH=src python -m pytest -q tests/unittests/opacity/diffgrid tests/unittests/utils/misc/grids_test.py` — 65 passed
- `git diff --check`
- parsed `pyproject.toml` successfully with Python 3.11

### CODE REVIEW

**Review by YuiKasagi — APPROVED** [review]
Thank you very much for the updates!

- I confirmed that all tests in `tests/unittests `pass in my environment.
- I also tried running `documents/tutorials/diffgrid_nuts_retrieval.ipynb`, and it worked well. I am thinking of incorporating this into my own retrieval code at some point.
- Here is my report on the CKD update: I compared magnitudes measured from spectra computed at sufficiently high spectral resolution with those obtained using CKD. Previously, I found differences of about 0.3 mag depending on the range of the temperature grid. With this update, however, the discrepancy has been reduced to around 0.02 mag(!). This gives me much more confidence in using CKD.

Image

**Comment by HajimeKawahara** [comment]
@YuiKasagi Thank you!

**Comment by astrostevanus** [comment]
Thank you for working on the updates! All of the unittests run smoothly in my environment after freshly installing this version of Exojax. I have also tried the new feature of diffgrid_nuts_retrieval.ipynb and it worked well with the old RTX A6000 (Ampere).

**Review by astrostevanus — APPROVED** [review]

**Review by KoHosokawa — APPROVED** [review]

**Review by YuiKasagi — APPROVED** [review]

**Review by HibikiYama — APPROVED** [review]
Thank you for the updates!
Here is my report:

- I can now generate spectral models without errors using the CKD tables rebuilt with v2.6.
- Changes to the limb-darkening calculation and a bug fix have made the retrieval about twice as fast.

**Review by ykawashima — APPROVED** [review]
Thank you for the updates, and I apologize for my late review. I also confirmed that the diffgrid retrieval tutorial works as expected.
https://secondearths.sakura.ne.jp/exojax/develop/tutorials/diffgrid_nuts_retrieval.html
